# Data Wrangling - Satander Value Prediction

### Data Preparation, Exploration, and Cleaning

Importing Package:

In [1]:
import pandas as pd
import numpy as np

Data Collection:

In [2]:
data_test = '/Users/sangth/Desktop/USF_Springboard/Capstone_2/Dataset/santander-value-prediction-challenge/test.csv'
data_train = '/Users/sangth/Desktop/USF_Springboard/Capstone_2/Dataset/santander-value-prediction-challenge/train.csv'

Read the dataset

In [3]:
df_test = pd.read_csv(data_test)
df_train = pd.read_csv(data_train)

Explore the dataset: shapes, summary statistics, types

In [4]:
print('Shape of the test csv dataset :', df_test.shape)
print('Shape of the train csv dataset :', df_train.shape)

print("\nSummary statistics:\n", df_test.describe())

Shape of the test csv dataset : (49342, 4992)
Shape of the train csv dataset : (4459, 4993)

Summary statistics:
           48df886f9     0deb4b6a8     34b15f335     a8cb14b00     2f0771a37  \
count  4.934200e+04  4.934200e+04  4.934200e+04  4.934200e+04  4.934200e+04   
mean   5.773787e+04  6.258726e+04  1.036752e+05  6.289853e+04  6.713354e+04   
std    1.745182e+06  2.322787e+06  2.586951e+06  2.765941e+06  3.206124e+06   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
50%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
75%    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
max    1.504447e+08  2.283295e+08  2.758171e+08  3.972621e+08  4.667591e+08   

          30347e683     d08d1fbe3     6ee66e115     20aa07010     dc5a8f1d8  \
count  4.934200e+04  4.934200e+04  4.934200e+04  4.934200e+04  4.934200e+04   
mean   8.083879e

In [5]:
df_test.head()

ID  48df886f9  0deb4b6a8  34b15f335  a8cb14b00  2f0771a37  \
0  000137c73        0.0        0.0        0.0        0.0        0.0   
1  00021489f        0.0        0.0        0.0        0.0        0.0   
2  0004d7953        0.0        0.0        0.0        0.0        0.0   
3  00056a333        0.0        0.0        0.0        0.0        0.0   
4  00056d8eb        0.0        0.0        0.0        0.0        0.0   

   30347e683  d08d1fbe3  6ee66e115  20aa07010  ...  3ecc09859  9281abeea  \
0        0.0        0.0        0.0        0.0  ...        0.0        0.0   
1        0.0        0.0        0.0        0.0  ...        0.0        0.0   
2        0.0        0.0        0.0        0.0  ...        0.0        0.0   
3        0.0        0.0        0.0        0.0  ...        0.0        0.0   
4        0.0        0.0        0.0        0.0  ...        0.0        0.0   

   8675bec0b  3a13ed79a  f677d4d13  71b203550  137efaa80  fb36b89d9  \
0        0.0        0.0        0.0        0.0        0.0        0.0   
1        0.0        0.0        0.0        0.0        0.0        0.0   
2        0.0        0.0        0.0        0.0        0.0        0.0   
3        0.0        0.0        0.0        0.0        0.0        0.0   
4        0.0        0.0        0.0        0.0        0.0        0.0   

   7e293fbaf  9fc776466  
0        0.0        0.0  
1        0.0        0.0  
2        0.0        0.0  
3        0.0        0.0  
4        0.0        0.0  

[5 rows x 4992 columns]

In [6]:
df_train.head()

ID      target  48df886f9  0deb4b6a8  34b15f335  a8cb14b00  \
0  000d6aaf2  38000000.0        0.0          0        0.0          0   
1  000fbd867    600000.0        0.0          0        0.0          0   
2  0027d6b71  10000000.0        0.0          0        0.0          0   
3  0028cbf45   2000000.0        0.0          0        0.0          0   
4  002a68644  14400000.0        0.0          0        0.0          0   

   2f0771a37  30347e683  d08d1fbe3  6ee66e115  ...  3ecc09859  9281abeea  \
0          0          0          0          0  ...        0.0        0.0   
1          0          0          0          0  ...        0.0        0.0   
2          0          0          0          0  ...        0.0        0.0   
3          0          0          0          0  ...        0.0        0.0   
4          0          0          0          0  ...        0.0        0.0   

   8675bec0b  3a13ed79a  f677d4d13  71b203550  137efaa80  fb36b89d9  \
0        0.0          0          0          0          0          0   
1        0.0          0          0          0          0          0   
2        0.0          0          0          0          0          0   
3        0.0          0          0          0          0          0   
4        0.0          0          0          0          0          0   

   7e293fbaf  9fc776466  
0          0          0  
1          0          0  
2          0          0  
3          0          0  
4          0          0  

[5 rows x 4993 columns]

In [7]:
df_test.info()
df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49342 entries, 0 to 49341
Columns: 4992 entries, ID to 9fc776466
dtypes: float64(4991), object(1)
memory usage: 1.8+ GB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4459 entries, 0 to 4458
Columns: 4993 entries, ID to 9fc776466
dtypes: float64(1845), int64(3147), object(1)
memory usage: 169.9+ MB


### Data Summary
The test dataset has 49342 rows and 4992 columns, while the train dataset has 4459 rows and 4993 columns. The majority of columns are integers or floats, with only one string column, the 'ID' column. The train dataset includes a 'target' column.


### Check for missing values

In [8]:
print("Missing Value counts in test dataset:", df_test.isnull().sum())
print("Missing Value counts in train dataset:", df_train.isnull().sum())

Missing Value counts in test dataset: ID           0
48df886f9    0
0deb4b6a8    0
34b15f335    0
a8cb14b00    0
            ..
71b203550    0
137efaa80    0
fb36b89d9    0
7e293fbaf    0
9fc776466    0
Length: 4992, dtype: int64
Missing Value counts in train dataset: ID           0
target       0
48df886f9    0
0deb4b6a8    0
34b15f335    0
            ..
71b203550    0
137efaa80    0
fb36b89d9    0
7e293fbaf    0
9fc776466    0
Length: 4993, dtype: int64


Both datasets do not have any missing values.

### Remove Constant Value Columns

Columns with constant values: Generally when we get problems with many columns, there might be few columns with constant value in train set.

In [9]:
unique_df_train = df_train.nunique().reset_index()
unique_df_train.columns = ["col_name", "unique_count"]
constant_df_train = unique_df_train[unique_df_train['unique_count']==1]
constant_df_train.shape


(256, 2)

In [10]:
unique_df_test = df_test.nunique().reset_index()
unique_df_test.columns = ["col_name", "unique_count"]
constant_df_test = unique_df_test[unique_df_test['unique_count']==1]
constant_df_test.shape

(0, 2)

There are 256 columns in the "train.csv" dataset that have constant values across all rows. These columns might not provide much information for modeling and analysis. (256 c, 2 = rows)

Remove constant values

In [11]:
# Get the list of constant column names
constant_column_names = constant_df_train['col_name'].tolist()
print(constant_column_names)

['d5308d8bc', 'c330f1a67', 'eeac16933', '7df8788e8', '5b91580ee', '6f29fbbc7', '46dafc868', 'ae41a98b6', 'f416800e9', '6d07828ca', '7ac332a1d', '70ee7950a', '833b35a7c', '2f9969eab', '8b1372217', '68322788b', '2288ac1a6', 'dc7f76962', '467044c26', '39ebfbfd9', '9a5ff8c23', 'f6fac27c8', '664e2800e', 'ae28689a2', 'd87dcac58', '4065efbb6', 'f944d9d43', 'c2c4491d5', 'a4346e2e2', '1af366d4f', 'cfff5b7c8', 'da215e99e', '5acd26139', '9be9c6cef', '1210d0271', '21b0a54cb', 'da35e792b', '754c502dd', '0b346adbd', '0f196b049', 'b603ed95d', '2a50e001c', '1e81432e7', '10350ea43', '3c7c7e24c', '7585fce2a', '64d036163', 'f25d9935c', 'd98484125', '95c85e227', '9a5273600', '746cdb817', '6377a6293', '7d944fb0c', '87eb21c50', '5ea313a8c', '0987a65a1', '2fb7c2443', 'f5dde409b', '1ae50d4c3', '2b21cd7d8', '0db8a9272', '804d8b55b', '76f135fa6', '7d7182143', 'f88e61ae6', '378ed28e0', 'ca4ba131e', '1352ddae5', '2b601ad67', '6e42ff7c7', '22196a84c', '0e410eb3d', '992e6d1d3', '90a742107', '08b9ec4ae', 'd95203ded'

In [12]:
# Drop columns with constant values from the DataFrame(df_train)
df_train_filtered = df_train.drop(columns=constant_column_names)
print("Shape of the filtered train dataset:", df_train_filtered.shape)

Shape of the filtered train dataset: (4459, 4737)


#### Handling Duplicates:
Duplicates can lead to biased analysis and modeling

In [13]:
duplicate_rows = df_train_filtered[df_train_filtered.duplicated()]
print("Number of Duplicate rows in train dataset:", duplicate_rows.shape[0])

Number of Duplicate rows in train dataset: 0


The train dataset does not contain any duplicate rows.

In [16]:
# Specify the path where you want to save the filtered train dataset
save_path = '/Users/sangth/Desktop/USF_Springboard/Capstone_2/Dataset/santander-value-prediction-challenge/filtered_train.csv'

# Save the filtered train dataset to a CSV file
df_train_filtered.to_csv(save_path, index=False)

# Confirm that the DataFrame has been saved successfully
print("Filtered train dataset saved to:", save_path)


Filtered train dataset saved to: /Users/sangth/Desktop/USF_Springboard/Capstone_2/Dataset/santander-value-prediction-challenge/filtered_train.csv
